pyrtl/rtllib/multipliers.py

```diff
@@ -172,18 +171,20 @@ def signed_tree_multiplier(A, B, reducer=adders.wallace_reducer, adder_func=adde
             bits[i + j].append(x & y)
 
     res = reducer(bits, bits_length+1, adder_func)
```

Comment:
Why do you need `bit_lengths + 1`? Also, the error is thrown earlier in the function anyways, so having the `trivial_mult` doesn't matter.